multiaddr/codecs/multihash.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import absolute_import
+
+import base58
+import six
+import varint
+
+from ..protocols import read_varint_code
+
+
+def to_bytes(proto, string):
+	# the address is a varint prefixed multihash string representation
+	try:
+		if six.PY2 and isinstance(string, unicode):
+			string = string.encode("ascii")
+		mm = base58.b58decode(string)
+	except Exception as ex:
+		raise ValueError("failed to parse p2p addr: %s %s" % (string, str(ex)))
+	size = varint.encode(len(mm))
+	if len(mm) < 5:
+		# TODO - port go-multihash so we can do this correctly
```

Comment:
While we could use https://github.com/multiformats/py-multihash here, I'm wondering whether there is any benefit in doing so: As far as `multiaddr` is concerned this is just an arbitrary base58-encoded binary string and there is nothing to be gained by parsing it as multihash other than performing basic validation really. Unless of course, `/p2p` is not a multihash, but a CID, but then we should stop calling it multihash.

Either way it should be renamed: If it is a base58-encoded multihash, we should call this `base58-str` and be done with it. If it is a CID, then this should be named `cid` and use the [`py-cid`](https://github.com/ipld/py-cid/) module.

Reply:
Agree. I don't think we should parse multihash either. From [go-multiaddr](https://github.com/multiformats/go-multiaddr/blob/152990b98bf9dce961ca2aefb1002cd96cfff6cb/transcoders.go#L250-L270), they just encode/decode base58 as well. What do you think we rename this 'p2p', the same as go-multiaddr?

Reply:
Done!